### PR TITLE
betMany questional optimization

### PR DESCRIPTION
**Pull request by BanzaiTokyo**

### CODE REVIEW

File: packages/hardhat/contracts/Lottery.sol [hunk 1]
```diff
@@ -93,10 +93,15 @@ contract Lottery is Ownable {
     /// @notice Calls the bet function `times` times
     function betMany(uint256 times) external {
         require(times > 0);
+        uint256 totalFees = betFee * times;
+        uint256 totalBetPrice = betPrice * times;
+        address[times] memory slotsToPush;
         while (times > 0) {
-            bet();
+            slotsToPush[times - 1] = msg.sender;
```
Comment (gurobokum) [inline packages/hardhat/contracts/Lottery.sol]: What array is it?

Comment (BanzaiTokyo) [reply packages/hardhat/contracts/Lottery.sol]: the idea was to prepare the array or addresses to push them to slots all at once. But it doesn't seem to be possible, the line 103 is invalid :( 
`_slots.push(slotsToPush);`

Comment (gurobokum) [reply packages/hardhat/contracts/Lottery.sol]: Got it, I think we may init slotsToPush based on `_slots`
And let's name it's as `slots` maybe?

File: packages/hardhat/contracts/Lottery.sol [hunk 2]
```diff
@@ -93,10 +93,15 @@ contract Lottery is Ownable {
...
         require(times > 0);
+        uint256 totalFees = betFee * times;
+        uint256 totalBetPrice = betPrice * times;
+        address[] memory slotsToPush = new uint256[](times);
         while (times > 0) {
-            bet();
+            slotsToPush[times - 1] = msg.sender;
             times--;
         }
+        _slots.push(slotsToPush);
```
Comment (gurobokum) [inline packages/hardhat/contracts/Lottery.sol]: Here you push array into array

**Pull request merged**